src/vm/eventpipeconfiguration.cpp

```diff
@@ -385,30 +424,28 @@ bool EventPipeConfiguration::RundownEnabled() const
     return m_rundownEnabled;
 }
 
-void EventPipeConfiguration::EnableRundown()
+void EventPipeConfiguration::EnableRundown(EventPipeSession *pSession)
```

Comment:
This function taking a session pointer seems odd to me. Mainly in that if that session is not configured properly (i.e. the rundown providers are not enabled, but some other random ones are) it seems that this function is run through just fine, but not really enable rundown as the name implies. What it would do is simply change the behavior to writing directly to the file rather than go through the buffer.

I think it would be more clear to do one of the following:
* Keep the session creation logic in this function and have it return a session pointer
* Have this function exclusively set m_rundownEnabled, and take no args, then add a function in EventPipe called something like `DoRundown` or `Rundown` that builds the rundown session object and enables rundown, enables the configuration with the session and so on
* Remove this function and RundownEnabled entirely and implement `EventPipe::DoRundown`, and have the m_rundownEnabled setting be a part of the sessions state

Reply:
Yeah, I think that this needs work as well.  When I implement CAPTURE_STATE I will make one of these changes.